apriltags_ros/include/apriltags_ros/apriltag_detector.h

```diff
@@ -33,30 +39,50 @@ class AprilTagDetector{
   ~AprilTagDetector();
  private:
   void enableCb(const std_msgs::Bool& msg);
-  void imageCb(const sensor_msgs::ImageConstPtr& msg,const sensor_msgs::CameraInfoConstPtr& cam_info);
+  void imageCb(const sensor_msgs::PointCloud2ConstPtr& depth_msg, const sensor_msgs::ImageConstPtr& rgb_msg_in,
+		const sensor_msgs::CameraInfoConstPtr& info_msg);
   std::map<int, AprilTagDescription> parse_tag_descriptions(XmlRpc::XmlRpcValue& april_tag_descriptions);
   bool getTransform(std::string t1, std::string t2, tf::Transform& output);
 
+  tf::Transform getDepthImagePlaneTransform(const sensor_msgs::PointCloud2ConstPtr& cloud,
+    std::pair<float,float> polygon[4], AprilTags::TagDetection& detection, tf::Vector3 x);
+
  private:
   std::map<int, AprilTagDescription> descriptions_;
   std::string sensor_frame_id_;
+
+  boost::shared_ptr<image_transport::ImageTransport> rgb_it_;
+
+  // Subscriptions
+  ros::NodeHandlePtr rgb_nh_;
   image_transport::ImageTransport it_;
-  image_transport::CameraSubscriber image_sub_;
+  message_filters::Subscriber<sensor_msgs::PointCloud2> sub_point_cloud_;
+  message_filters::Subscriber<sensor_msgs::CameraInfo> sub_info_;
+
+  image_transport::SubscriberFilter sub_rgb_;
+
+  typedef ApproximateTime<sensor_msgs::PointCloud2, sensor_msgs::Image, sensor_msgs::CameraInfo> SyncPolicy;
+  typedef message_filters::Synchronizer<SyncPolicy> Synchronizer;
+  boost::shared_ptr<Synchronizer> sync_;
+
   image_transport::Publisher image_pub_;
+  ros::Publisher plane_cloud_pub_;
   ros::Publisher detections_pub_;
   ros::Publisher pose_pub_;
+  ros::Publisher plane_pose_pub_;
   ros::Subscriber enable_sub_;
   tf::TransformBroadcaster tf_pub_;
   boost::shared_ptr<AprilTags::TagDetector> tag_detector_;
   bool projected_optics_;
   bool enabled_;
+  float plane_inlier_threshold_;
+  float plane_angle_threshold_;
```

Comment:
Might be worth adding a comment with the units.  Is angle in radians, degrees, or something else?  And if the angle is in degrees instead of radians, make sure to call it out.  Most of ros uses radians instead of degrees for everything.